aws/endpoints/v3model.go

```diff
@@ -74,16 +74,29 @@ func (p partition) canResolveEndpoint(service, region string, strictMatch bool)
 	return p.RegionRegex.MatchString(region)
 }
 
+func allowLegacyEmptyRegion(service string) bool {
+	legacy := map[string]struct{}{
+		"ec2metadata": {},
+	}
```

Comment:
Does it help to keep the legacy map outside this function to avoid creating it every time this function is called?

Reply:
updated to represent this better. moved map into a package variable.